compiler/optimizations/gpuTransforms.cpp

```diff
@@ -969,24 +978,55 @@ static void outlineGPUKernels() {
 }
 
 static void logGpuizableLoops() {
-  forv_Vec(BlockStmt, block, gBlockStmts) {
-    if (ForLoop* forLoop = toForLoop(block)) {
-      if (forLoop->isOrderIndependent())
-        if (GpuizableLoop(forLoop).isEligible())
-          if (debugPrintGPUChecks)
-            printf("Found viable forLoop %s:%d[%d]\n",
-                   forLoop->fname(), forLoop->linenum(), forLoop->id);
-    } else if (CForLoop* forLoop = toCForLoop(block)) {
-      if (GpuizableLoop(forLoop).isEligible())
-        if (debugPrintGPUChecks)
-          printf("Found viable CForLoop %s:%d[%d]\n",
-                 forLoop->fname(), forLoop->linenum(), forLoop->id);
+  struct LocationComparator {
+    bool operator()(const CForLoop* item1, const CForLoop* item2) const {
+      const char* s1 = item1->fname();
+      const char* s2 = item2->fname();
+      int result = strcmp(s1, s2);
+      if (result != 0) {
+        return result < 0;
+      }
+      return item1->linenum() < item2->linenum();
     }
+  };
+  std::set<CForLoop*, LocationComparator> eligibleLoops;
+  std::set<CForLoop*, LocationComparator> ineligibleLoops;
+
+  forv_Vec(FnSymbol*, fn, gFnSymbols) {
+    std::vector<BaseAST*> asts;
+    collect_asts(fn, asts);
```

Comment:
There's no `collectCForLoops`? Maybe we should make one.

Reply:
Sure, I've added `collectCForLoopStmts`.

Though now I'm wondering why we don't just have a generic function that would let me do something like: `collectAst<CForLoop>(fn, loops)`